src/StorageSync/StorageSync/CloudEndpoint/NewCloudEndpointCommand.cs

```diff
@@ -192,6 +193,24 @@ public override void ExecuteCmdlet()
                     throw new PSArgumentException(nameof(StorageAccountResourceId));
                 }
 
+                if(this.IsParameterBound(c=>c.StorageAccountTenantId))
+                {
+                    if(StorageAccountTenantId != DefaultContext.Tenant.Id)
+                    {
+                        throw new PSArgumentException(string.Format(StorageSyncResources.NewCloudEndpointCrossTenantErrorFormat, StorageAccountTenantId, DefaultContext.Tenant.Id));
+                    }
+                }
+
+                if(storageAccountResourceIdentifier.Subscription != DefaultContext.Subscription.Id)
```

Comment:
is it possible this line will be executed when no subscription under this tenant?

Reply:
Shouldn't be this already covered in Microsoft.Azure.Commands.ResourceManager.Common ?

Reply:
protected override IAzureContext DefaultContext
        {
            get
            {
                if (DefaultProfile == null || DefaultProfile.DefaultContext == null || DefaultProfile.DefaultContext.Account == null)
                {
                    throw new PSInvalidOperationException(Resources.RunConnectAccount);
                }

                return DefaultProfile.DefaultContext;
            }
        }

Reply:
this will cover the case that user was not login, if there are no subscriptions under current tenant, "DefaultContext.Subscription" will be null referenced

Reply:
please ignore this comment  if current tenant cannot have no subscription.